bottlerocket/agents/src/bin/vsphere-k8s-cluster-resource-agent/vsphere_k8s_cluster_provider.rs

```diff
@@ -0,0 +1,763 @@
+        .output()
+        .context(*resources, "Failed to launch govc process")?;
+    // Tag the VM template with the OS tag
+    let tag_attach_output = Command::new("govc")
+        .args(["tags.attach", "os:bottlerocket"])
+        .arg(&vm_full_path)
+        .output()
+        .context(*resources, "Failed to launch govc process")?;
+    if !tag_attach_output.status.success() {
+        return Err(ProviderError::new_with_context(
+            *resources,
+            format!(
+                "Failed to tag VM template '{}': {}",
+                vm_template_name,
+                String::from_utf8_lossy(&tag_attach_output.stderr)
+            ),
+        ));
+    }
+    // Create the 'eksdRelease' tag category
+    Command::new("govc")
+        .args([
+            "tags.category.create",
+            "-m",
+            "-t",
+            "VirtualMachine",
+            "eksdRelease",
+        ])
+        .output()
+        .context(*resources, "Failed to launch govc process")?;
+    // Create the 'eksRelease' tag, e.g 'eksRelease:kubernetes-1-23-eks-6'
+    // EKS-A needs to make sure the VM template is tagged with a compatible EKS-D release number.
+    // Finding out which EKS-D release we're using in a given Bottlerocket OVA is hard and
+    // we really really don't care which EKS-D release for a given K8s cluster version we're
+    // testing here, we're just going to tag the VM template with a range of release numbers.
+    // The tag just has to exist for EKS-A to be happy. We choose 50 since it's highly unlikely
+    // a given K8s minor version will have 50 EKS-D releases.
+    let k8s_ver_str = config
+        .version
+        .context(*resources, "K8s version missing from configuration")?
+        .major_minor_without_v()
+        .replace('.', "-");
+    for release_number in 1..50 {
+        let eksd_ver_str = format!("kubernetes-{}-eks-{}", k8s_ver_str, release_number);
+        // Same reasons as above, we don't need to check the status here
+        Command::new("govc")
+            .args(["tags.create", "-c", "eksdRelease"])
+            .arg(&format!("eksdRelease:{}", eksd_ver_str))
+            .output()
+            .context(*resources, "Failed to launch govc process")?;
+        // Tag the VM template with the EKSD tag
+        let tag_attach_output = Command::new("govc")
+            .args(["tags.attach", &format!("eksdRelease:{}", eksd_ver_str)])
+            .arg(&vm_full_path)
+            .output()
+            .context(*resources, "Failed to launch govc process")?;
+        if !tag_attach_output.status.success() {
+            return Err(ProviderError::new_with_context(
+                *resources,
+                format!(
+                    "Failed to tag VM template '{}': {}",
+                    vm_template_name,
+                    String::from_utf8_lossy(&tag_attach_output.stderr)
+                ),
+            ));
+        }
+        sleep(Duration::from_millis(200));
```

Comment:
suspicious. You probably want to call out why this is here and how arbitrary/non-arbitrary 200 ms is.

Reply:
I didn't want to overload the vsphere API with create/attach tag calls. The 200ms is completely unnecessary. I'll experiment with his a bit more and add a comment.

Reply:
I typo'd "arbitrary" into "unnecessary". Thanks brain.
Going to try without the sleep and test.

Reply:
It's fine to have it if it makes you feel more confident in reliability. You can just a comment to say why you think it might be helpful. No need to be overly scientific with the value.